### PR TITLE
remove locations-config from required args

### LINKED ISSUE #314: Make locations-config an optional argument for kmd
Currently the `kmd` CLI requires that the user creates a `locations.yml` file (it can have any name) with contents like:

```
locations:
  foo: my-computer.foo.example.com
```

This sets the value of `ERT_LSF_SERVER` in the scripts `local` and `local.csh`, which has content like:

```shell
host=$(hostname)
location=$(echo $host | cut -d "-" -f 1)

case ${location} in
   foo)
      export ERT_LSF_SERVER=my-computer.foo.example.com
        ;;
esac
```

The tool **requires** this file, even though it is probably not useful for most other users of the tool (hypothetical users at the moment, but...)

If we do change this, the README will need changing too.

### PR DESCRIPTION
**Pull request by kwinkunks**
resolves #314

### CODE REVIEW

**Comment by hnformentin** [comment]
Would it be possible to add tests for this new behaviour?

File: komodo/cli.py [hunk 1]
```diff
@@ -88,13 +88,16 @@ def _main(args):
...
 
-    with open(args.locations_config, mode="r", encoding="utf-8") as defs, open(
-        Path(args.release) / "local", mode="w", encoding="utf-8"
-    ) as local_activator, open(
-        Path(args.release) / "local.csh", mode="w", encoding="utf-8"
-    ) as local_csh_activator:
-        defs = yml.safe_load(defs)
-        local.write_local_activators(data, defs, local_activator, local_csh_activator)
+    if args.locations_config is not None:
+        with open(args.locations_config, mode="r", encoding="utf-8") as defs, open(
```
Comment (berland) [inline komodo/cli.py:92]: This  `with` construct is just so hard to read, when all it is doing is to read one file and write two others. But maybe a rewrite can be done in a different PR.

Comment (kwinkunks) [reply komodo/cli.py:92]: I agree... TBH I didn't dare touch it and didn't want to dig into `komodo.local`. I made an issue for it.

File: komodo/cli.py [hunk 2]
```diff
@@ -244,6 +241,12 @@ def cli_main():
     optional_args.add_argument("--postinst", "-P", type=str)
 
     optional_args.add_argument("--renamer", "-R", default="rename", type=str)
+    optional_args.add_argument(
+        "--locations-config",
+        type=str,
+        required=False,
+        help="Path to locations.yml, a map of location to a server.",
```
[the remark this answers is not in the document]
Comment (kwinkunks) [reply komodo/cli.py]: Good one, done.

**Comment by berland** [comment]
Should we have a integration test that certifies that no other part of the code assumes that there is a location config?

**Comment by kwinkunks** [comment]
Good points about testing. Added a test to ensure that we can run `kmd` without the `--locations-config` argument and that it does not produce a `local` script.

**Review by berland — APPROVED** [review]
looking good!

**Comment by kwinkunks** [comment]
:thinking: Hm, tests pass on local but fail on CI. Investigating...

**Comment by kwinkunks** [comment]
:information_source: the problem was that `virtualenv` was not in the environment. Rather than trying to fix this, possibly getting into `bootstrap.sh` and other magic, I instead made my test more like the existing test in `test_cli.py`. This is okay, but has the disadvantage of having to add one more argument (`--extra-data-dirs`) to the "minimal" example.

**Pull request merged**